### PR TITLE
VM ingestion

### PR DESCRIPTION
**Pull request by shwstppr**
Documentation PR for VM Ingestion feature.
Feature PR: https://github.com/apache/cloudstack/pull/3606

### CODE REVIEW

File: source/adminguide/virtual_machines.rst [hunk 1]
```diff
@@ -227,6 +227,16 @@ To create a VM from an ISO:
...
 
+Importing VMs
+-------------
+
+Virtual machines that have been created out of band, directly from hypervisor can be imported in CloudStack(currently only for VMware). Import action can only be performed by administrator and only using API. Following APIs can be used while importing unmanaged virtual machines:
+
+-  listUnmanagedInstances - to list unmanaged virtual machines for a cluster.
+
+-  importUnmanagedInstance - to import an unmanaged virtual machine into CloudStack.
+
```
Comment (rhtyd) [inline source/adminguide/virtual_machines.rst]: @shwstppr add details on how to use the `.py` script etc?

Comment (shwstppr) [reply source/adminguide/virtual_machines.rst]: @rhtyd yes, will be adding that. This was just initial commit to create PR, working on rest of the documentation

**Comment by shwstppr** [comment]
@andrijapanicsb @PaulAngus Can you please review changes
I'm not sure if it is the correct location in documentation to add vm ingestion notes so please advise changes needed.

**Comment by PaulAngus** [comment]
requesting docbuild

**Comment by shwstppr** [comment]
requesting docbuild

**Comment by PaulAngus** [comment]
requesting docbuild

**Comment by PaulAngus** [comment]
requesting docbuild

File: source/adminguide/virtual_machines/VM_Ingestion.rst [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+   resize of the disk when importing. The disk will remain with its original size, but CloudStack will have a
+   record as per the offering.
+
+Host and Storage Tags
+#####################
+
+When the **migrateallowed** parameter is set to true, if the host or storage tags in the compute/disk offerings are incompatible with the current host and/or
+storage pool(s), CloudStack will migrate the VM and its volumes to a suitable host and storage pool.
+
+When migrateallowed is false and there is a conflict, an appropriate error will be returned.
```
Comment (shwstppr) [inline source/adminguide/virtual_machines/VM_Ingestion.rst]:
```suggestion
When **migrateallowed** is false and there is a conflict, an appropriate error will be returned.
```

File: source/adminguide/virtual_machines/VM_Ingestion.rst [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+.. note::
+   To use Custom Unconstrained type of compute offering, CPU speed will need to be passed using details parameter when the CPU reservation is not set for
+   the unmanaged VM in vSphere. CPU speed in the latter case can be passed as, details[0].cpuSpeed=SOME_VALUE.
+
+
+Disk Offerings
+###############
+
+
+To import a VM which has data disks attached, a map of the disk ID and corresponding disk offering ID must be passed via the *datadiskofferinglist* parameter.
```
[the remark this answers is not in the document]
Comment (PaulAngus) [reply source/adminguide/virtual_machines/VM_Ingestion.rst]: oh boy, what happens in code?  the diskoffering is a messy subject, a user doesn't choose a disk offering for the root, it's linked to the serviceoffering ID

[the remark this answers is not in the document]
Comment (shwstppr) [reply source/adminguide/virtual_machines/VM_Ingestion.rst]: While importing it uses disk offeering passedc with param `diskofferingid`. It is a required parameter for importUnmanagedInstance API

[the remark this answers is not in the document]
Comment (PaulAngus) [reply source/adminguide/virtual_machines/VM_Ingestion.rst]: hmm. if you consider creating a VM from a template, you don't pass a disk offering, it's implicit from the service offering...

**Comment by shwstppr** [comment]
requesting docbuild

**Comment by andrijapanicsb** [comment]
@shwstppr Can you please update this one:
- there is conflict
- it mentioned Backup and Recovery, but there is actually a separate B&R PR https://github.com/apache/cloudstack-documentation/pull/85
- can you sync with Rohit and see what is what and to have this one (VM ingestion) merged.

**Comment by PaulAngus** [comment]
if you merge B&R, the commits will match.
B&R was used as the basis for this to avoid conflicts

**Comment by andrijapanicsb** [comment]
@PaulAngus are you staying that this one is reviewed and LGTM?

**Comment by PaulAngus** [comment]
@shwstppr it doesn't look like you've updated this PR after review by Steve Roles ..?

**Comment by shwstppr** [comment]
@PaulAngus I don't think Steve ever reviewed this one. There was no draft word doc for it. After initial updates it was mostly updated by you.

**Comment by shwstppr** [comment]
@andrijapanic will resolve the conflict tomorrow morning

**Comment by shwstppr** [comment]
@andrijapanicsb I've fixed conflicts. cc @PaulAngus

**Comment by andrijapanicsb** [comment]
requesting docbuild

**Comment by andrijapanicsb** [comment]
@shwstppr @PaulAngus - can you advise if this is good to go or not, we need it asap for 4.14.
thx

File: source/adminguide/virtual_machines/VM_Ingestion.rst [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+From ACS 4.14 onwards, CloudStack is able to list these VMs via the listUnmanagedInstances API command and then import (also known as ingest)
+those unmanaged VMs via the importUnmanagedInstance API so that they become CloudStack managed guest instances
+
+.. note:: This is currently only available for **vSphere** clusters.
+
+
+Use Cases and General Usage
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ability to import VMs allows Cloud operators (both public and private) on onboard new tenants simply and quickly,
```
Comment (andrijapanicsb) [inline source/adminguide/virtual_machines/VM_Ingestion.rst:35]:
```suggestion
The ability to import VMs allows Cloud operators (both public and private) to onboard new tenants simply and quickly,
```

**Comment by andrijapanicsb** [comment]
pinging again for LGTM if possible.

**Comment by andrijapanicsb** [comment]
Merging as it is, looks LGTM in general, but we can always improve it if needed.
Required for 4.14 to go out.

**Pull request merged**